iceoryx_posh/include/iceoryx_posh/popo/untyped_publisher.hpp

```diff
@@ -25,9 +25,13 @@ namespace iox
 {
 namespace popo
 {
-template <typename base_publisher_t = BasePublisher<>>
+template <typename H = mepoo::NoCustomHeader, typename base_publisher_t = BasePublisher<>>
 class UntypedPublisherImpl : public base_publisher_t
```

Comment:
Instead of having this as template parameter, it would also be possible to do it like with the C API and have a method like.
```
template <typename H>
bool configureCustomHeader() noexcept;
```
Alternatively, it could also be put into the publisher options on the C++ side but then there are options which would be discarded with the typed API.

There will be an additional feature with a hook to access the custom header e.g. on sample delivery which could be a callback with the signature
```
void(const ChunkHeader*, H*)
```
With the current approach, this can be made typesafe.

Reply:
Personally I like the template param approach. Easy and clean. I think it doesn't belong to the publisher options. They are described in the re-written examples as "QoS options". We should be aligned with DDS implementations here.

Reply:
This is not user API but I suppose the ser will only have to provide the template parameter if a non-default header is desired. Should be ok and the added type-safety is a plus.